tests/components/emoncms/test_config_flow.py

```diff
@@ -138,6 +138,6 @@ async def test_options_flow_failure(
     await setup_integration(hass, config_entry)
     result = await hass.config_entries.options.async_init(config_entry.entry_id)
     await hass.async_block_till_done()
-    assert result["errors"]["base"] == "failure"
+    assert result["errors"]["base"] == EMONCMS_FAILURE["message"]
```

Comment:
Better to assert "api_error", and assert that the description placeholder has the corresponding error message.

Reply:
```suggestion
    assert result["errors"]["base"] == "api_error"
    assert result["description_placeholders"]["details"] == "failure"
```

Reply:
OK thanks for the tip. It is indeed really better like that.